### ISSUE TITLE
Olimex ESP32-POE(-ISO) Ethernet & Wifi at the same time

### ISSUE THREAD

**Issue by sipodd** [esphome/feature-requests#2102]
**Describe the problem you have/What new integration you would like**

Possibility to enable Ethernet, Wifi and Bluetooth, on this particular Boards at the same time.

**Please describe your use case for this integration and alternatives you've tried:**

While the Olimex ESP32-POE seems to be capable, it is not possible to enable Ethernet Component and Wifi Component at the same time within ESPHome.
ref: https://github.com/OLIMEX/ESP32-POE/tree/master/SOFTWARE/ARDUINO/ESP32_PoE_Ethernet_WIFI_test

**Additional context**

I would like to use Olimex ESP32-POE Boards as Wifi AP while the devices are connected and powered via POE-Ethernet. (so: using Ethernet as an uplink; the ESP32-POE as a bridge)
If there is enough memory left, using it as a Bluetooth proxy would be greate, but not the main concern of this FR.

**Comment by nagyrobi** [comment]
This is not specific to Olimex ESP32-POE. ESPHome doesn't support parallel usage of Ethernet and Wifi, regardless of the board:
https://esphome.io/components/ethernet.html

**Comment by sipodd** [comment]
> This is not specific to Olimex ESP32-POE. ESPHome doesn't support parallel usage of Ethernet and Wifi, regardless of the board: https://esphome.io/components/ethernet.html

I know, thats the reason for this FR...i'd like to have it supported.

**Comment by JeroenVanOort** [comment]
Using an ESP32 as an ethernet backed wifi access point would require more than just being able to use the two components simultaneously, as traffic would need to be routed between the two interfaces. Although I couldn't help you with the specifics because I simply don't know, I believe it would take quite some work to add this to esphome.

Out of curiosity, I looked for ESP32 based AP's outside of esphome and while there are some projects, I don't have the impression that your use case is a common one. From this I take that people have found other solutions to be better suited. Unless you want a hobby project to spend many hours on, I would advise you to buy a (second hand) AP and use a [PoE splitter](https://letmegooglethat.com/?q=poe+splitter+gigabit+12v). From the 12V jack, you can (using some DIY) feed both the AP and the ESP32. Most AP's also serve as an ethernet switch, so the ESP could take it's connectivity from there. This is what I use for PoE powering some (OpenWrt based) AP's that don't support PoE directly.

**Comment by mdrobniu** [comment]
Is there a way just only to enable wifi association. i.e. I'd like to open my garadge door once esp home completes a 4-way handshake with my carplay.

**Comment by kpfleming** [comment]
I too would like to use both Ethernet and WiFi simultaneously, but not as an AP or bridge... just regular 'client' connectivity on both interfaces.

**Comment by ssieb** [comment]
> +1 Bluetooth and Ethernet at the same time needed

@Grandma-Betty that's already possible.  This is about wifi.

**Comment by nagyrobi** [comment]
Works for me with the same board... what Bluetooth devices are you trying to use?

**Comment by nagyrobi** [comment]
> Are you using the same code for ethernet?

Yes, and use a static IP address with `manual_ip` and `use_address` for higher stability.

**Comment by ssieb** [comment]
If you use `manual_ip`, you don't need `use_address`.

**Comment by kpfleming** [comment]
I suspect it does not need to be 'synchronized to an NTP server', it just needs to have the wall-clock time set in some fashion. Using the `homeassistant` time platform should be sufficient as well.

**Comment by Adminius** [comment]
To bring this FR to the life again:
I would also use WIFI and wired LAN "simultaneously".
Like if wired LAN not connected use WiFi or vice versa

**Comment by Gunni** [comment]
The implementation of this should be fairly simple, just use standard routing tables, but with a higher local preference for ethernet, etc.

This should be generalizable, just do like normal OSes do!. Longest prefix wins, send packets with the source ip for the route that wins.

examples:
1. Just wifi:
  ```
  default via fe80::ffff dev wifi0 metric 30
  default via 192.0.1.254 dev wifi0 metric 30
  192.0.1.0/24 dev wifi0 scope link metric 30
  2001:db8::/64 dev wifi0 metric 30
  ```

2. Wifi and wired
  ```
  default via fe80::ffff dev wired0 metric 10
  default via 192.0.1.254 dev wired0 metric 10
  192.0.1.0/24 dev wired0 scope link metric 10
  2001:db8::/64 dev wired0 metric 10

  default via fe80::ffff dev wifi0 metric 30
  default via 192.0.1.254 dev wifi0 metric 30
  192.0.1.0/24 dev wifi0 scope link metric 30
  2001:db8::/64 dev wifi0 metric 30
  ```

3. You can keep adding stuff like thread, wireguard, etc!

The nics don't need to end on the same broadcast domain, and can have wildly different routes on each interface, just remember that only one route wins each routing decision.

p.s. another thing that needs doing is getting rid of "connection timeout" for the wired submodule, the concept of "connection timeout" when waiting for link-up on ethernet is really dumb.

**Comment by aznel** [comment]
Implemented upstream. [esphome/esphome#14255](https://github.com/esphome/esphome/pull/14255) lifted CONFLICTS_WITH = ["wifi"] on ethernet: and added a network: priority: list — merged to dev 2026-07-23, first released in 2026.8.0 (2026-08-19). Default-route arbitration followed in [esphome/esphome#17797](https://github.com/esphome/esphome/pull/17797).

network:
  priority: [ethernet, wifi]   # must name every configured interface
ethernet: ...
wifi: ...

Suggest closing. While open it's a top search result for "esphome ethernet wifi simultaneously" and reads as unimplemented.

**Comment by Gunni** [comment]
Why not openthread too? Or a generic networking route table?
